src/IdentityServer4/src/Validation/Default/DeviceAuthorizationRequestValidator.cs

```diff
@@ -70,13 +70,13 @@ private DeviceAuthorizationRequestValidationResult Invalid(ValidatedDeviceAuthor
 
         private void LogError(string message, ValidatedDeviceAuthorizationRequest request)
         {
-            var requestDetails = new DeviceAuthorizationRequestValidationLog(request);
+            var requestDetails = new DeviceAuthorizationRequestValidationLog(request, _options.Logging.TokenRequestSensitiveValuesFilter);
```

Comment:
The name of the property is TokenRequestSensitiveValuesFilter but it's being used in the DeviceAuthorizationRequestValidationLog. Could this be confusing for someone?

Reply:
Isn't that in essence also a token request? Do you have a better idea?

Reply:
Hmm, the way I understood device authorization endpoint was to authorize a client application and to get back as a result, a user code, a device code, and verification url (and other fields) that a user can then use in a browser to authenticate themselves while the client application pokes the token endpoint.
So my perception is it's not really a token request from what I understand. It's really a device authorization request.

An alternative would be to add another property in the LoggingOptions named DeviceAuthorizationRequestSensitiveValuesFilter with the clientId and client secret.

And yet another alternative, which I think might be the best, is to leave the DeviceAuthorizationRequestValidationLog as is (i.e. remove the changes that were done) and keep the fixed list of sensitive values defined in the class. This request is not typically open for customization like the token request is.

Reply:
Agreed. thanks!